include/simdjson/inline/document_stream.h

```diff
@@ -160,6 +164,7 @@ inline void document_stream::next() noexcept {
 #endif
     if (error) { continue; } // If the error was EMPTY, we may want to load another batch.
     // Run stage 2 on the first document in the batch
+    doc_index = batch_start + parser.implementation->structural_indexes[parser.implementation->next_structural_index];
```

Comment:
Here the parser attempts to parse the document a second time and, sure enough, we reset doc_index.